### PR TITLE
Add support for communication with cli

### PR DESCRIPTION
**Pull request by cooligus**

### CODE REVIEW

**Review by maxidragon — APPROVED** [review]
Please wait for Bartek's review before merging

File: src-tauri/src/main.rs [hunk 1]
```diff
@@ -1,8 +1,17 @@
 // Prevents additional console window on Windows in release, DO NOT REMOVE!!
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
 
+use tauri::Manager;
+
+mod spito_helper;
+
```
Comment (BaderBC) [inline src-tauri/src/main.rs:6]:
```suggestion

#[cfg(target_os = "linux")]
```

We should add it in order to allow compilation only for linux

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -1,8 +1,17 @@
 // Prevents additional console window on Windows in release, DO NOT REMOVE!!
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
```
Comment (BaderBC) [inline src-tauri/src/main.rs]:
```suggestion
```

It isn't necessary because we support only linux

File: src-tauri/src/spito_helper.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+use std::process::{Command};
+use dbus_crossroads::Crossroads;
+use dbus::blocking::Connection;
+use tauri::{AppHandle, Manager};
+
+#[tauri::command]
+pub(crate) fn start_spito_cli(ruleset: &str, rule: &str) -> Result<(), String> {
```
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]:
```suggestion
pub fn start_spito_cli(ruleset: &str, rule: &str) -> Result<(), String> {
```

File: src-tauri/src/spito_helper.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+const DBUS_OBJECT_PATH: &str = "/org/avorty/spito/gui";
+
+#[derive(Clone, serde::Serialize)]
+struct Payload {
+    mess_type: String,
+    message: String,
+}
+
+#[tauri::command]
+pub(crate) fn start_spito_server(app: AppHandle) -> Result<(), String>{
```
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]:
```suggestion
pub fn start_spito_server(app: AppHandle) -> Result<(), String> {
```

File: src-tauri/src/main.rs [hunk 5]
```diff
@@ -1,8 +1,17 @@
...
-    .run(tauri::generate_context!())
-    .expect("error while running tauri application");
+    tauri::Builder::default()
+        .setup(|app| {
+            spito_helper::start_spito_server(app.app_handle())?;
+            Ok(())
+        })
+        .invoke_handler(tauri::generate_handler![spito_helper::start_spito_cli])
+        .run(tauri::generate_context!())
+        .expect("error while running tauri application");
```
Comment (BaderBC) [inline src-tauri/src/main.rs]:
```suggestion
        .expect("error while running spito-gui application");
```

File: src-tauri/src/spito_helper.rs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        .arg(rule)
+        .spawn().map_err(|err| err.to_string())?;
+    Ok(())
+}
+
+const DBUS_ID: &str = "org.avorty.spito.gui";
+const DBUS_OBJECT_PATH: &str = "/org/avorty/spito/gui";
+
+#[derive(Clone, serde::Serialize)]
+struct Payload {
```
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]: Maybe some more meaningful like e.g. `DbusPayload`

**Review by BaderBC — CHANGES_REQUESTED** [review]

**Review by BaderBC — CHANGES_REQUESTED** [review]
Don't listen for `Echo` and add support for `api.info.*`

**Comment by cooligus** [comment]
> Don't listen for `Echo` and add support for `api.info.*`

`Echo` is much better than creating 5 function that does completely same stuff

**Comment by BaderBC** [comment]
> > Don't listen for `Echo` and add support for `api.info.*`
> 
> 
> 
> `Echo` is much better than creating 5 function that does completely same stuff

When a rule calls `api.info.log` it should push other part of gui than other methods. e.g. `api.info.error` could be shown as a red popup, while `api.info.log` could be hidden somewhere

**Comment by cooligus** [comment]
> > > Don't listen for `Echo` and add support for `api.info.*`
> > 
> > 
> > `Echo` is much better than creating 5 function that does completely same stuff
> 
> When a rule calls `api.info.log` it should push other part of gui than other methods. e.g. `api.info.error` could be shown as a red popup, while `api.info.log` could be hidden somewhere

whether it's error or something else specifies first arguemnt of `echo`

File: src-tauri/src/spito_helper.rs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        .spawn().map_err(|err| err.to_string())?;
+    Ok(())
+}
+
+const DBUS_ID: &str = "org.avorty.spito.gui";
+const DBUS_OBJECT_PATH: &str = "/org/avorty/spito/gui";
+
+#[derive(Clone, serde::Serialize)]
+struct DBusEchoPayload {
+    mess_type: String,
```
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]: Make it camel case when serialized (you can use serde decorator for this) and consider using full name

File: src-tauri/src/spito_helper.rs [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    });
+}
+
+fn _start_spito_server(app: AppHandle) -> anyhow::Result<()> {
+    let conn = Connection::new_session()?;
+    conn.request_name(DBUS_ID, false, true, false)?;
+
+    let mut cr = Crossroads::new();
+    let token = cr.register(DBUS_ID, |builder| {
+        builder.method("Echo", ("mess_type", "message"), (), move |_ctx, _cr, (mess_type, message, ): (String, String, )| {
```
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]: What about renaming `Echo` to `Info` or `api.info`

File: src-tauri/src/spito_helper.rs (hunk 7 above)
Comment (cooligus) [reply src-tauri/src/spito_helper.rs:22]: RustRover says it should be snake case

**Review by BaderBC — APPROVED** [review]
LGTM!

**Pull request merged**